.github/workflows/ci.yaml

```diff
@@ -0,0 +1,85 @@
+name: MonoKickstart Ci
+on: [push]
+jobs:
+  build:
+    
+    name: Steam Runtime
+    runs-on: ubuntu-20.04
+    container: registry.gitlab.steamos.cloud/steamrt/sniper/sdk:beta
+    # TODO: Run on Rocky Linux, using buntu right now because I haven't checked the package names on RHEL yet
+    steps:
+      - uses: actions/checkout@v4
+        with:
+            submodules: true
+            path: monokickstart
+
+      - name: Build libgdiplus
+        run: |
+            sudo apt update
+            sudo apt install libgif-dev autoconf libtool automake build-essential gettext libglib2.0-dev libcairo2-dev libtiff-dev libexif-dev ninja-build mono-complete -y
+            git clone https://gitlab.winehq.org/mono/libgdiplus --depth=1 --recursive --shallow-submodules
+            cd libgdiplus
+            ./autogen.sh
+            make -j${nproc}
+
+      - name: Upload libgdiplus Artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: libgdiplus
+            path: libgdiplus/src/.libs/libgdiplus.so.0.0.0
+
+      - name: Build mono
+        run: |
+            git clone https://gitlab.winehq.org/mono/mono --depth=1 --recursive --shallow-submodules
+            cd mono
+            git apply -v ../monokickstart/patches/mono*
+            cd external/corefx
+            git apply -v ../../../monokickstart/patches/corefx*
+            cd ../corert
+            git apply -v ../../../monokickstart/patches/corert*
+            cd ../.. 
+            ./autogen.sh --with-ikvm-native=no
+            make -j${nproc}
+            find . | grep -i Mono.Posix.dll
+
+      - name: Upload libMonoPosixHelper Artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: libMonoPosixHelper
+            path: mono/support/.libs/libMonoPosixHelper.so
+
+      - name: Upload net_4_x-linux Artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: net_4_x-linux
+            path: mono/mcs/class/lib/net_4_x-linux/
+
+      - name: Build MonoKickstart
+        run: |
+            cd monokickstart
+            cmake -B build -D CMAKE_BUILD_TYPE=Release -G Ninja
+            cmake --build build --parallel
+
+      - name: Upload kick.bin.x86_64 Artifact
+        uses: actions/upload-artifact@v4
+        with:
+            name: kick.bin.x86_64
+            path: |
+              monokickstart/build/kick.bin.x86_64

```

Comment:
The files are right but it doesn't flatten the directories like we'd want, so we get to do some tedious copypasting for that...